### ISSUE TITLE
[a11y]: ComposedModal - accessibility checker throws 'Element with "region" role does not have a label' violation

### ISSUE THREAD

**Issue by makafsal** [carbon-design-system/carbon#19512]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

v1.83.0

### React version

v18.3.1

### Automated testing tool and ruleset

IBM Equal access accessibility checker - latest

### Description

When the `ModalBody` has scrollable content, the accessibility checker throws the violation:

![Image](https://github.com/user-attachments/assets/aace2672-d113-4d29-8c99-1ce83219f9c5)

To resolve this issue, if I've given `aria-label` to the `ModalBody` component, then it is resolved. But when there is no scroll, it shows the following violation:

![Image](https://github.com/user-attachments/assets/2bd07e11-b2c3-4107-984c-828ea542833e)

Note: According to [Chrome focus on scrollers](https://developer.chrome.com/blog/keyboard-focusable-scrollers) update, we don't need to provide `tabindex='0'` explicitly anymore.

### Reproduction/example

https://stackblitz.com/edit/xwynwbht?file=src%2FApp.jsx

### Steps to reproduce

1. Go to the [stackblitz](https://stackblitz.com/edit/xwynwbht?file=src%2FApp.jsx)
2. Remove the `aria-label` from `ModalBody` for now
3. Shrink the screen to make a scrollable content (if there is no scroll in the body)
4. Scan using IBM accessibility checker
5. Now we can see the first violation
6. To fix this first violation, provide `aria-label` back to the `ModalBody` component
7. Again, run the check, and we can see that the first violation is gone
8. Now, reduce the inspector height and give more space for `Modal` and remove scrolls.
9. Again, run the the checker
10. Now we can see the second violation

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by jclintonibm** [comment]
We are seeing a similar issue `The tabbable element's role 'region' is not a widget role`, location:
```
<div
    role="region"
    tabindex="0"
    class="cds--layer-two cds--modal-content cds--modal-scroll-content"
    aria-label="build settings">
```

**Comment by chubbywhale** [comment]
I am also affected by the `The tabbable element's role 'region' is not a widget role` issue. 

Is there plan to address this in near future?

**Comment by caritaou** [comment]
I am seeing this issue as well, reproducible in storybook:

<img width="1670" height="950" alt="Image" src="https://github.com/user-attachments/assets/486712fa-833f-490a-90f3-1b3aea0f4f15" />

**Comment by caritaou** [comment]
The Cloud Pak for Data team needs this fix in order to be fully compliant for accessibility. Please provide the status on who is assigned to this issue and an ETA for the fix. Any questions, you can contact Debby Nakamura, Release Manager for Cloud Pak for Data.

**Comment by habeeb-mir** [comment]
@heloiselui Hi, could you help us triage this? Thanks!
